### PR TITLE
Adding multidimio

### PR DESCRIPTION
**Pull request by tasansal**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tasansal** [comment]
@conda-forge/help-python, ready for review!

**Comment by tasansal** [comment]
@conda-forge-admin, please ping team

Hello! we are open-sourcing our first library at my company, and we hope to have this available by Aug 30th. We would be very grateful if we can get this reviewed and approved before then! If not, that's also ok, I just wanted to ask.

Thank you very much for supporting the community! It has been a lot of fun publishing our first conda-forge package.

File: recipes/multidimio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - mdio --help
+  requires:
+    - pip
+
+about:
+  home: https://mdio.dev
+  summary: Cloud-native, scalable, and user-friendly multi dimensional energy data!
+  license: Apache-2.0
+  license_file:
+    - LICENSE
```
Comment (BastianZim) [inline recipes/multidimio/meta.yaml:50]:
```suggestion
    - LICENSE
  dev_url: https://github.com/TGSAI/mdio-python
```

File: recipes/multidimio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - mdio = mdio.__main__:main
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8,<3.11
+    - pip
+    - poetry
```
Comment (BastianZim) [inline recipes/multidimio/meta.yaml:23]:
```suggestion
    - poetry-core
```

Ref: https://github.com/TGSAI/mdio-python/blob/429a6cbf7dcfa5037f0491952e150a532ca9b8ea/pyproject.toml#L109

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, then this is ready to go 🚀

**Review by BastianZim — APPROVED** [review]

**Comment by tasansal** [comment]
@BastianZim, thanks a lot!

**Pull request merged**